src/corner.rs

```diff
@@ -60,16 +59,20 @@ impl Corner {
                     }) {
                         last_event = Some(event);
                     } else {
-                        debug!(
-                            "Ignored the event due to too fast after unlock."
-                        );
+                        debug!("Ignored the event due to too fast after unlock.");
                     }
                 }
                 Err(_error) => {
-                    if last_event
-                        .map_or(false, |value| value == CornerEvent::Enter)
-                    {
-                        self.execute_command()?;
+                    if last_event.map_or(Ok(false), |value| -> Result<bool> {
+                        if value == CornerEvent::Enter {
+                            self.execute_enter_command()?;
+                        } else if value == CornerEvent::Leave {
+                            self.execute_exit_command()?;
+                        } else {
+                            return Ok(false);
+                        }
+                        return Ok(true);
+                    })? {
                         command_done_at = Some(Instant::now());
                     }
```

Comment:
I would avoid calling the execute methods in a mapping in an if statement. I would want to avoid side-effects in an if statement. It's also a bit hard to read.

```suggestion
                    if let Some(event) = last_event {
                        if event == CornerEvent::Enter {
                            self.execute_enter_command()?;
                        } else if value == CornerEvent::Leave {
                            self.execute_exit_command()?;
                        }
                        command_done_at = Some(Instant::now());
                    }
```